openhtf/exe/__init__.py

```diff
@@ -122,40 +131,113 @@ def _RunTestLoop(self):
       # Top level steps required to run a single iteration of the Test.
       _LOG.info('Starting test %s', self.test.code_info.name)
 
-      # Save exit_stack on self so we can access it when Stop() is called.
-      self._current_exit_stack = exit_stack
-      exit_stack.callback(lambda: setattr(self, '_current_exit_stack', None))
-
-      plug_manager = self._MakePlugManager(suppressor)
-      executor = self._MakePhaseExecutor()
-
+      # Any access to self._exit_stack must be done while holding this lock.
+      with self._lock:
+        # Save exit_stack on self so we can access it if Stop() is called.
+        self._exit_stack = exit_stack
+
+        # Reset some stateful attributes to None when we're done with all other
+        # teardown.
+        exit_stack.callback(lambda: setattr(self, '_exit_stack', None))
```

Comment:
good call, done.  this kind of blew up on me, largely from overly wordy comments (that were as much for internal justification as aiding the next person in understanding the code), reduced that verbosity a bit too.